File: binding/src/libqstylefactory.cpp
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QStyle>
#include <QStyleFactory>
#include <qstylefactory.h>
#include "libqstylefactory.h"
#include "libqstylefactory.hxx"

QStyleFactory* QStyleFactory_new(const QStyleFactory* other) {
	 return new QStyleFactory(*other);
}

QStyleFactory* QStyleFactory_new2(QStyleFactory* other) {
	 return new QStyleFactory(*other);
}

QStyleFactory* QStyleFactory_new3() {
	 return new QStyleFactory();
}

QStyleFactory* QStyleFactory_new4(const QStyleFactory* param1) {
	 return new QStyleFactory(*param1);
}

void QStyleFactory_CopyAssign(QStyleFactory* self, QStyleFactory* other) {
    *self = *other;
}

void QStyleFactory_MoveAssign(QStyleFactory* self, QStyleFactory* other) {
    *self = std::move(*other);
}

libqt_list QStyleFactory_Keys() {
	return QStyleFactory::keys();
}

QStyle* QStyleFactory_Create(const libqt_string param1) {
	return QStyleFactory::create(QString::fromUtf8(param1.data, param1.len));
}

void QStyleFactory_OperatorAssign(QStyleFactory* self, const QStyleFactory* param1) {
	self->operator=(*param1);
}

void QStyleFactory_Delete(QStyleFactory* self) {
    delete self;
}

File: binding/include/libqstylefactory.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qstylefactory.html
pub const qstylefactory = struct {

    /// New constructs a new QStyleFactory object.
    pub fn New(other: ?*anyopaque) QtC.QStyleFactory {
        return qtc.QStyleFactory_new(@ptrCast(other));
    }


    /// New2 constructs a new QStyleFactory object.
    pub fn New2(other: ?*anyopaque) QtC.QStyleFactory {
        return qtc.QStyleFactory_new2(@ptrCast(other));
    }


    /// New3 constructs a new QStyleFactory object.
    pub fn New3() QtC.QStyleFactory {
        return qtc.QStyleFactory_new3();
    }


    /// New4 constructs a new QStyleFactory object.
    pub fn New4(param1: ?*anyopaque) QtC.QStyleFactory {
        return qtc.QStyleFactory_new4(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QStyleFactory_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QStyleFactory_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn Keys() []const u8 {
        return qtc.QStyleFactory_Keys();
    }

    pub fn Create(param1: []const u8) QtC.QStyle {
        const param1_str = qtc.libqt_string{
    .len = param1.len,
    .data = param1.ptr,
};
return qtc.QStyleFactory_Create(param1_str);
    }

    pub fn OperatorAssign(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QStyleFactory_OperatorAssign(@ptrCast(self), @ptrCast(param1));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QStyleFactory_Delete(@ptrCast(self));
    }
};
